README.md

```diff
@@ -0,0 +1,175 @@
+# Open Context for Node.js
+
+This is a simple library for providing Distributed Context Problem
+in Node.js programs.
+
+DCP is most beneficial in large service oriented architectures. Each request
+has some associated metadata, and we want to make sure that data stays with
+the tree of RPCs that service the root request.
+
+The `Context` object in this library stores that request metadata, or baggage.
+It's intended to be implemented into RPC libraries, who would be responsible
+for actually serializing the data onto the wire.
+
+It also supports 'joining', a process where we copy all values from one
+context to another.
+
+A `Context` object is essentially just a bag of key/value pairs. By having a
+common interface, we can support `Context` in different RPC libraries.
+
+## `Context()`
+
+The `Context` object stores arbitrary key/value pairs. The constructor creates
+a new blank context.
+
+### `Context#set(key, value)`
+
+Maps `key` to `value` within this context.
+
+### `Context#get(key)`
+
+Gets the value of `key`.
+
+### `Context#keys()`
+
+Gets the list of keys.
+
+### `Context#joinWith(context, mergeFns)`
```

Comment:
The idea of carrying the merge functions on the context itself may need some play time, but my gut is that the RPC library should add mergeFns to the context so the caller’s joinWith will work without expressly threading.